frontend/src/hooks/useSessionValid.js

```diff
@@ -34,6 +35,8 @@ function useSessionValid() {
   const { setAlertDetails } = useAlertStore();
   const navigate = useNavigate();
   const userSession = useUserSession();
+  const logout = useLogout();
+  const timestamp = new Date().getTime();
```

Comment:
@tahierhussain One doubt, will the timestamp be regenerated each time a request is made?
Or will it happen only once when this component is mounted the first time? If so, the timestamp will remain the same thus effectively having no difference with not having a timestamp at all.

Reply:
@hari-kuriakose With our current approach, we call the hook method only once when a component is mounted, so it shouldn't be a problem. However, as a safeguard, I have moved the timestamp inside the function. This way, a new timestamp will be generated even if the component re-renders and requires calling the function again.